src/order/RelayOrder.ts

```diff
@@ -0,0 +1,307 @@
+        actions,
+      ],
+    ] = decoded;
+    return new RelayOrder(
+      {
+        reactor,
+        swapper,
+        nonce,
+        deadline: deadline.toNumber(),
+        inputs: inputs.map(
+          ([token, recipient, startAmount, maxAmount]: [
+            string,
+            number,
+            number,
+            string,
+            boolean
+          ]) => {
+            return {
+              token,
+              startAmount,
+              maxAmount,
+              recipient,
+            };
+          }
+        ),
+        decayStartTime: decayStartTime.toNumber(),
+        decayEndTime: decayEndTime.toNumber(),
+        actions: actions,
+      },
+      chainId,
+      permit2
+    );
+  }
+
+  /**
+   * @inheritdoc order
+   */
+  toJSON(): RelayOrderInfoJSON & {
+    permit2Address: string;
+    chainId: number;
+  } {
+    return {
+      chainId: this.chainId,
+      permit2Address: this.permit2Address,
+      reactor: this.info.reactor,
+      swapper: this.info.swapper,
+      nonce: this.info.nonce.toString(),
+      deadline: this.info.deadline,
+      actions: this.info.actions,
+      decayStartTime: this.info.decayStartTime,
+      decayEndTime: this.info.decayEndTime,
+      inputs: this.info.inputs.map((input) => ({
+        token: input.token,
+        startAmount: input.startAmount.toString(),
+        maxAmount: input.maxAmount.toString(),
+        recipient: input.recipient,
+      })),
+    };
+  }
+
+  /**
+   * @inheritdoc order
+   */
+  serialize(): string {
+    const abiCoder = new ethers.utils.AbiCoder();
+    return abiCoder.encode(RELAY_ORDER_ABI, [
+      [
+        [
+          this.info.reactor,
+          this.info.swapper,
+          this.info.nonce,
+          this.info.deadline,
+        ],
+        this.info.inputs.map((input) => [
+          input.token,
+          input.recipient,
+          input.startAmount,
+          input.maxAmount,
+        ]),
+        this.info.decayStartTime,
+        this.info.decayEndTime,
+        this.info.actions,
+      ],
+    ]);
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  getSigner(signature: SignatureLike): string {
+    return ethers.utils.computeAddress(
+      ethers.utils.recoverPublicKey(
+        SignatureTransfer.hash(
+          this.toPermit(),
+          this.permit2Address,
+          this.chainId,
+          this.witness()
+        ),
+        signature
+      )
+    );
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  permitData(): PermitBatchTransferFromData {
+    return SignatureTransfer.getPermitData(
+      this.toPermit(),
+      this.permit2Address,
+      this.chainId,
+      this.witness()
+    ) as PermitBatchTransferFromData;
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  hash(): string {
+    return ethers.utils._TypedDataEncoder
+      .from(RELAY_WITNESS_TYPES)
+      .hash(this.witnessInfo());
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  resolve(options: OrderResolutionOptions): ResolvedRelayOrder {
```

Comment:
how does this compile given ResolvedRelayOrder !== ResolvedOrder?

Reply:
A bit concerned that integrators relying on the `Order` interface ie FE may get mixed up by this mistyping

Reply:
yea agree, it was compiling from the type union but removing that so each class implements resolve separately